codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/EventStreamMarshallTestCases.kt

```diff
@@ -46,163 +54,188 @@ internal object EventStreamMarshallTestCases {
-                .some_int(5)
-                .some_string("hello")
-                .build()
-            );
-            let result = ${format(generator)}().marshall(event);
-            assert!(result.is_ok(), "expected ok, got: {:?}", result);
-            let message = result.unwrap();
-            let headers = headers_to_map(message.headers());
-            assert_eq!(&str_header("event"), *headers.get(":message-type").unwrap());
-            assert_eq!(&str_header("MessageWithNoHeaderPayloadTraits"), *headers.get(":event-type").unwrap());
-            assert_eq!(&str_header(${testCase.requestContentType.dq()}), *headers.get(":content-type").unwrap());
-
-            validate_body(
-                message.payload(),
-                ${testCase.validMessageWithNoHeaderPayloadTraits.dq()},
-                MediaType::from(${testCase.requestContentType.dq()})
-            ).unwrap();
-            """,
-        )
+        unitTest("message_with_blob") {
+            rustTemplate(
+                """
+                let event = TestStream::MessageWithBlob(
+                    MessageWithBlob::builder().data(#{BlobInput:W}).build()
+                );
+                let result = $generator::new().marshall(event);
+                assert!(result.is_ok(), "expected ok, got: {:?}", result);
+                let message = result.unwrap();
+                let headers = headers_to_map(message.headers());
+                assert_eq!(&str_header("event"), *headers.get(":message-type").unwrap());
+                assert_eq!(&str_header("MessageWithBlob"), *headers.get(":event-type").unwrap());
+                assert_eq!(&str_header("application/octet-stream"), *headers.get(":content-type").unwrap());
+                assert_eq!(&b"hello, world!"[..], message.payload());
+                """,
+                "BlobInput" to builderInput("Blob::new(&b\"hello, world!\"[..])"),
+            )
+        }
+
+        unitTest("message_with_string") {
+            rustTemplate(
+                """
+                let event = TestStream::MessageWithString(
+                    MessageWithString::builder().data(#{StringInput}).build()
+                );
+                let result = $generator::new().marshall(event);
+                assert!(result.is_ok(), "expected ok, got: {:?}", result);
+                let message = result.unwrap();
+                let headers = headers_to_map(message.headers());
+                assert_eq!(&str_header("event"), *headers.get(":message-type").unwrap());
+                assert_eq!(&str_header("MessageWithString"), *headers.get(":event-type").unwrap());
+                assert_eq!(&str_header("text/plain"), *headers.get(":content-type").unwrap());
+                assert_eq!(&b"hello, world!"[..], message.payload());
+                """,
+                "StringInput" to builderInput("\"hello, world!\""),
+            )
+        }
+
+//        unitTest( "message_with_struct") {
```

Comment:
Is this one intentionally commented out?

Reply:
Uncommented in https://github.com/awslabs/smithy-rs/pull/2342/commits/a14463aa90bc2a90c75d62dafd46c5009795fd21.